### PR TITLE
gh-107017: removed mention that C does it the same way

### PR DESCRIPTION
**Pull request by JakubDotPy**
Don't assume the reader knows the C language, especially in the tutorial.

* Issue: gh-107017

----
:books: Documentation preview :books:: https://cpython-previews--107020.org.readthedocs.build/

### CODE REVIEW

File: Doc/tutorial/controlflow.rst [hunk 1]
```diff
@@ -4,8 +4,8 @@
 More Control Flow Tools
 ***********************
 
-Besides the :keyword:`while` statement just introduced, Python uses the usual
-flow control statements known from other languages, with some twists.
+Besides the :keyword:`while` statement just introduced, Python uses a few more
```
Comment (evildmp) [inline Doc/tutorial/controlflow.rst]: "As well as" would be more idiomatic English than "besides".

File: Doc/tutorial/controlflow.rst [hunk 2]
```diff
@@ -163,10 +163,10 @@ arguments.  In chapter :ref:`tut-structures`, we will discuss in more detail abo
 :keyword:`!break` and :keyword:`!continue` Statements, and :keyword:`!else` Clauses on Loops
 ============================================================================================
 
-The :keyword:`break` statement, like in C, breaks out of the innermost enclosing
+The :keyword:`break` statement breaks out of the innermost enclosing
 :keyword:`for` or :keyword:`while` loop.
 
-Loop statements may have an :keyword:`!else` clause; it is executed when the loop
+The :keyword:`for` loop statements may have an extra :keyword:`!else` clause; it is executed when the loop
```
Comment (evildmp) [inline Doc/tutorial/controlflow.rst]: I think the line length needs to be checked here.

**Review by evildmp — CHANGES_REQUESTED** [review]

File: Doc/tutorial/controlflow.rst [hunk 3]
```diff
@@ -4,8 +4,8 @@
 More Control Flow Tools
 ***********************
 
-Besides the :keyword:`while` statement just introduced, Python uses the usual
-flow control statements known from other languages, with some twists.
+Besides the :keyword:`while` statement just introduced, Python uses a few more
+that will be introduced in this chapter.
```
Comment (evildmp) [inline Doc/tutorial/controlflow.rst]: "... that we will encounter..."

File: Doc/tutorial/controlflow.rst [hunk 4]
```diff
@@ -163,14 +163,14 @@ arguments.  In chapter :ref:`tut-structures`, we will discuss in more detail abo
...
 
-Loop statements may have an :keyword:`!else` clause; it is executed when the loop
-terminates through exhaustion of the iterable (with :keyword:`for`) or when the
-condition becomes false (with :keyword:`while`), but not when the loop is
-terminated by a :keyword:`break` statement.  This is exemplified by the
-following loop, which searches for prime numbers::
+The :keyword:`for` loop statements may have an extra :keyword:`!else` clause;
+it is executed when the loop terminates through exhaustion of the iterable
+(with :keyword:`for`) or when the condition becomes false (with :keyword:`while`),
+but not when the loop is terminated by a :keyword:`break` statement.
```
Comment (evildmp) [inline Doc/tutorial/controlflow.rst]: I find the original language here rather hard to parse because the concepts are quite unfamiliar (a loop that terminates through exhaustion of an iterable...).

Perhaps:

A for or while loop can include an else clause. 

In a for loop, the else clause is executed after the loop reaches its final iteration. In a while loop, it's executed after the loop's condition becomes false. 

In either kind of loop, the else clause is **not** executed if the loop was terminated by a break.

File: Doc/tutorial/controlflow.rst [hunk 5]
```diff
@@ -163,14 +163,14 @@ arguments.  In chapter :ref:`tut-structures`, we will discuss in more detail abo
...
+This is exemplified by the following loop, which searches for prime numbers::
```
Comment (evildmp) [inline Doc/tutorial/controlflow.rst]: exemplified _in_

**Review by evildmp — CHANGES_REQUESTED** [review]

File: Doc/tutorial/controlflow.rst [hunk 6]
```diff
@@ -166,7 +166,7 @@ arguments.  In chapter :ref:`tut-structures`, we will discuss in more detail abo
 The :keyword:`break` statement breaks out of the innermost enclosing
 :keyword:`for` or :keyword:`while` loop.
 
-Loop statements may have an :keyword:`!else` clause; it is executed when the loop
+The :keyword:`for` loop statements may have an extra :keyword:`!else` clause; it is executed when the loop
```
Comment (pochmann) [inline Doc/tutorial/controlflow.rst]: This just seems bad. You restricted it to `for` statements, but it also applies to `while` statements. And the `else` clause isn't really "extra" (I even checked some dictionaries, they all defined it like *"added to an existing or usual amount or number; additional. Example: they offered him an extra thirty-five cents an hour"*. And loop statements can only have *one* `else` clause.)

**Review by evildmp — APPROVED** [review]
